### ISSUE TITLE
[v3] YouTube oembed returning consent.youtube.com link

### ISSUE THREAD

**Issue by kinglozzer** [oscarotero/Embed#434]
This isn’t really an issue with this library (at least, I’m 99% sure it isn’t) so I’ll close this straight away - I just wanted to post it here for visibility, in case someone else like me is searching for this problem and also wondering if they’re alone...

`https://youtube.com/oembed` is sometimes returning an iframe as expected, and sometimes returning an anchor which links to `https://consent.youtube.com`:

```html
<a href="https://consent.youtube.com/ml?continue=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3D<VIDEOIDHERE>%26feature%3Dyoutu.be&amp;gl=GB&amp;hl=en-GB&amp;pc=yt&amp;uxe=24020388&amp;src=1">
    Before you continue to YouTube
</a>
```

I can’t recreate this locally, or even reliably on the live site, so it’s really difficult to try to debug what’s happening here. Given that the consent URL wants to redirect to the video, and not `https://youtube.com/oembed`, I can only assume that it is actually reaching the oembed API and that is returning the consent URL. Requesting the same embed code again later seems to work fine, so my best guess is that it depends which of YouTube’s edge servers responds.

I’ve reported it to YouTube via the “Send Feedback” thing as that’s the only place I could find to report an issue. I have no idea if I should expect a response!

**Comment by kinglozzer** [comment]
Re-opening as it’s now happening more widely for people (in the EU anyway). Note that this only appears to be happening on v3 of the library, not sure if it’s patchable or if this will just be a “wontfix”.

The problem is that when embedding a URL like `https://www.youtube.com/watch?v=YgGzAKP_HuM`, v3 of the library will first attempt to fetch that URL itself before it calls the oembed. The original URL redirects to the consent page:
```
https://consent.youtube.com/m?continue=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DYgGzAKP_HuM&gl=GB&m=0&pc=yt&uxe=23983172&hl=en-GB&src=1
```

The library then tries to call the oembed endpoint with that URL as the source, instead of the original video URL:
```
https://www.youtube.com/oembed?url=https%3A%2F%2Fconsent.youtube.com%2Fml%3Fcontinue%3Dhttps%253A%252F%252Fwww.youtube.com%252Fwatch%253Fv%253DYgGzAKP_HuM%26gl%3DGB%26hl%3Den-GB%26pc%3Dyt%26uxe%3D23983172%26src%3D1&format=json
```

A patch for this would ideally ensure that the oembed endpoint is called with the _original_ URL. I’ve not investigated how much work that’s likely to be

**Comment by oscarotero** [comment]
Ok, pull request merged and v3.4.14 published.
Thanks

**Comment by kinglozzer** [comment]
Thanks @oscarotero! Unfortunately there’s one thing I didn’t take into consideration: when people use short `youtu.be` URLs instead of the full URL - I’ll submit a new PR for that, it should be a pretty simple tweak
